### PR TITLE
POC: PagedPagination

### PR DESCRIPTION
**Pull request by rcatlin**
This PR

* [x] Implement `PagedPagination`

### CODE REVIEW

**Comment by greydnls** [comment]
This allows you to pass in a per-page limit in the query parameters, which isn't something we currently have documented as supporting, also, it doesn't allow anyway to override the default if you don't send in that limit via request. 

I think it's probably also a good idea to change the query parameter to "per-page" or similar, rather than limit

File: src/Middleware/Request/Pagination/PagedPagination.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use League\Pipeline\StageInterface;
+use League\Route\Http\Exception\BadRequestException;
+use Refinery29\Piston\Middleware\GetOnlyStage;
+use Refinery29\Piston\Middleware\Payload;
+use Refinery29\Piston\Request;
+
+class PagedPagination implements StageInterface
+{
+    use SinglePagination;
+    use GetOnlyStage;
```
Comment (rcatlin) [inline src/Middleware/Request/Pagination/PagedPagination.php]: I need to remove his unused trait.

**Comment by rcatlin** [comment]
> This allows you to pass in a per-page limit in the query parameters, which isn't something we currently have documented as supporting

Our API-Standards currently document using `page` and `limit` for page-based pagination.
https://github.com/refinery29/api-standards#implementing-page-based-pagination

> it doesn't allow anyway to override the default if you don't send in that limit via request.

I'll fix that.

> I think it's probably also a good idea to change the query parameter to "per-page" or similar, rather than limit

I like the idea of not using `limit` with page-based pagination. I can update our documentation to use `perPage` or `per-page`.

File: src/Middleware/Request/Pagination/PagedPagination.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+namespace Refinery29\Piston\Middleware\Request\Pagination;
+
+use League\Pipeline\StageInterface;
+use League\Route\Http\Exception\BadRequestException;
+use Refinery29\Piston\Middleware\Payload;
+use Refinery29\Piston\Request;
+
+class PagedPagination implements StageInterface
```
Comment (greydnls) [inline src/Middleware/Request/Pagination/PagedPagination.php]: What about nullable constructor argument for this class to over-write default pag?

Comment (rcatlin) [inline src/Middleware/Request/Pagination/PagedPagination.php]: Just added it. Love the idea.

**Comment by greydnls** [comment]
:+1:

**Pull request merged**